### PR TITLE
Refactor mark analyses deleted

### PR DESCRIPTION
**Pull request by henrikstranneheim**
## Description

Refactor function mark analysis as deleted.

### Added

- Tests
- Added `get_analyses_for_case` function

### Changed

- Refactored and migrated `mark_analyses_deleted` into CRUD modules.
- Refactored `case_name` to `case_id`

### How to prepare for test
- [ ] ssh to hasta (depending on type of change)
- [ ] activate stage: `us`
- [ ] request trailblazer-stage on hasta: `paxa`
- [ ] install on stage:
    ```Shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_trailblazer -t trailblazer -b refactor_mark_analyses_deleted -a
    ```
- [ ] ssh to clinical-db (depending on type of change)
- [ ] install on stage:
`bash /home/proj/production/servers/resources/clinical-db.scilifelab.se/update-trailblazer-ui-stage.sh [THIS-BRANCH-NAME]`

### How to test
- [ ] login to ...
- [ ] do ...

### Expected test outcome
- [ ] check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Review by diitaz93 — COMMENTED** [review]
Looks great! 💯

**Review by diitaz93 — APPROVED** [review]

File: trailblazer/server/api.py [hunk 1]
```diff
@@ -235,13 +235,17 @@ def post_delete_analysis():
...
-    content = request.json
-    old_analyses = store.mark_analyses_deleted(case_id=content.get("case_id"))
-    data = [stringify_timestamps(analysis_obj.to_dict()) for analysis_obj in old_analyses]
-    if data:
-        return jsonify(*data), 201
-    return jsonify(None), 201
+    """Mark all analysis belonging to a case as deleted."""
+    post_request: Response.json = request.json
+    case_analyses: Optional[List[Analysis]] = store.update_case_analyses_as_deleted(
+        case_name=post_request.get("case_id")
```
[the remark this answers is not in the document]
Comment (henrikstranneheim) [reply trailblazer/server/api.py:241]: Also moving to case_name instead of case_id.

[the remark this answers is not in the document]
Comment (henrikstranneheim) [reply trailblazer/server/api.py:241]: I think the use of name or id is difficult right now and we should try to come up with a definition of when to use what.

Maybe one should use id if it guaranteed to be unique. If not name is more appropriate @Clinical-Genomics/sysdev ? What do you think 🤔?

[the remark this answers is not in the document]
Comment (henrikstranneheim) [reply trailblazer/server/api.py]: Went with this as I think it at least provides a clear distinction between the two. `case_id `it is.

**Comment by henrikstranneheim** [comment]
### CLI and API ✅ 
```
62° [henrik.stranneheim@hasta:~] [S_base] 2m5s $ trailblazer scan
Status in Tower: richurchin - 278293 - 4bBxPTmgBJ7PSR
Updated status richurchin - 278293: qc
Error updating analysis for: case - finerrhino : PermissionError
Error updating analysis for: case - immuneamoeba : PermissionError
Error updating analysis for: case - modestakita : PermissionError
Error updating analysis for: case - famouscorgi : PermissionError
Error updating analysis for: case - topskite : PermissionError
Error updating analysis for: case - firstviper : PermissionError
Error updating analysis for: case - equalthrush : PermissionError
Error updating analysis for: case - openduck : PermissionError
Error updating analysis for: case - busyturkey : PermissionError
Error updating analysis for: case - strongsponge : PermissionError
Error updating analysis for: case - helpedegret : PermissionError
Error updating analysis for: case - nativeloon : PermissionError
Error updating analysis for: case - supercougar : PermissionError
Error updating analysis for: case - socialcrab : PermissionError
Error updating analysis for: case - activelion : PermissionError
Error updating analysis for: case - amusedspider : PermissionError
Error updating analysis for: case - greatbear : PermissionError
Error updating analysis for: case - aceviper : PermissionError
Error updating analysis for: case - brightape : PermissionError
Error updating analysis for: case - wittyghoul : PermissionError
Error updating analysis for: case - safeshiner : PermissionError
Error updating analysis for: case - brightegret : PermissionError
Error updating analysis for: case - fitloon : PermissionError
Error updating analysis for: case - decentcalf : PermissionError
Error updating analysis for: case - knownburro : PermissionError
Error updating analysis for: case - smoothlocust : PermissionError
Error updating analysis for: case - cuteprawn : PermissionError
Error updating analysis for: case - frankpigeon : PermissionError
Error updating analysis for: case - politeowl : PermissionError
Error updating analysis for: case - likedmutt : PermissionError
Error updating analysis for: case - wholegull : PermissionError
Error updating analysis for: case - promptowl : PermissionError
Error updating analysis for: case - nicewhale : PermissionError
Error updating analysis for: case - activedodo : PermissionError
Error updating analysis for: case - guidedmole : PermissionError
Error updating analysis for: case - busyturtle : PermissionError
Error updating analysis for: case - caringdrum : PermissionError
Error updating analysis for: case - lovedcaiman : PermissionError
Error updating analysis for: case - awakemarten : PermissionError
Error updating analysis for: case - neededbass : PermissionError
Error updating analysis for: case - divinesawfly : PermissionError
Error updating analysis for: case - modestoriole : PermissionError
Error updating analysis for: case - onekoi : PermissionError
Error updating analysis for: case - immunefawn : PermissionError
Error updating analysis for: case - manycow : PermissionError
Error updating analysis for: case - heroicmule : PermissionError
Error updating analysis for: case - inorca : PermissionError
Error updating analysis for: case - protahr : PermissionError
Error updating analysis for: case - trustysquid : PermissionError
Error updating analysis for: case - solidhusky : PermissionError
Error updating analysis for: case - moralbeagle : PermissionError
Error updating analysis for: case - cracktiger : PermissionError
Error updating analysis for: case - daringrat : PermissionError
Error updating analysis for: case - nativepossum : PermissionError
Error updating analysis for: case - savingstag : FileNotFoundError
Error updating analysis for: case - HW2V2DRX2 : FileNotFoundError
Error updating analysis for: case - divinetomcat : PermissionError
Error updating analysis for: case - handyshrimp : FileNotFoundError
Error updating analysis for: case - setcolt : PermissionError
Error updating analysis for: case - proudfish : PermissionError
Error updating analysis for: case - funnyaphid : FileNotFoundError
Error updating analysis for: case - onestag : PermissionError
Error updating analysis for: case - newgopher : PermissionError
Error updating analysis for: case - helpedlab : MissingSqueueOutput
Error updating analysis for: case - goodbengal : MissingSqueueOutput
Error updating analysis for: case - grandaphid : PermissionError
All analyses updated!
```

Screenshot 2023-09-05 at 08 18 33

```
θ60° [henrik.stranneheim@hasta:~] [S_base] 15s $ trailblazer delete  --cancel-jobs --force  278296
Cancelling job 5089936 - bwa_mem2
Cancelling job 5089935 - bwa_mem2
Cancelling job 5089937 - samtools_merge
Cancelling job 5089938 - samtools_merge
Cancelling job 5089939 - samtools_merge
Cancelling job 5089940 - markduplicates
Cancelling job 5089941 - markduplicates
Cancelling job 5089942 - markduplicates
Cancelling job 5089968 - cnvnator_ar
Cancelling job 5089969 - cnvnator_ar
Cancelling job 5089970 - cnvnator_ar
Cancelling job 5089972 - manta
Cancelling job 5089976 - sv_varianteffectpredictor
Cancelling job 5089977 - sv_vcfparser
Cancelling job 5089978 - sv_rankvariant
Cancelling job 5089991 - deepvariant
Cancelling job 5089992 - deepvariant
Cancelling job 5089993 - deepvariant
Cancelling job 5089994 - glnexus_merge
Cancelling job 5090010 - prepareforvariantannotationblock
Cancelling job 5090011 - rhocall_ar
Cancelling job 5090012 - bcftools_norm
Cancelling job 5090013 - variant_annotation
Cancelling job 5090023 - frequency_filter
Cancelling job 5090024 - cadd_ar
Cancelling job 5090025 - varianteffectpredictor
Cancelling job 5090026 - vcfparser_ar
Cancelling job 5090027 - rankvariant
Cancelling job 5089943 - chanjo_sexcheck
Cancelling job 5089944 - chanjo_sexcheck
Cancelling job 5089945 - chanjo_sexcheck
Cancelling job 5089946 - samtools_subsample_mt
Cancelling job 5089947 - samtools_subsample_mt
Cancelling job 5089948 - samtools_subsample_mt
Cancelling job 5089949 - tiddit_coverage
Cancelling job 5089950 - tiddit_coverage
Cancelling job 5089951 - tiddit_coverage
Cancelling job 5089952 - chromograph_cov
Cancelling job 5089953 - chromograph_cov
Cancelling job 5089954 - chromograph_cov
Cancelling job 5089955 - mitodel
Cancelling job 5089956 - mitodel
Cancelling job 5089957 - mitodel
Cancelling job 5089958 - sambamba_depth
Cancelling job 5089959 - sambamba_depth
Cancelling job 5089960 - sambamba_depth
Cancelling job 5089961 - smncopynumbercaller
Cancelling job 5089962 - picardtools_collectmultiplemetrics
Cancelling job 5089963 - picardtools_collectmultiplemetrics
Cancelling job 5089964 - picardtools_collectmultiplemetrics
Cancelling job 5089965 - picardtools_collecthsmetrics
Cancelling job 5089966 - picardtools_collecthsmetrics
Cancelling job 5089967 - picardtools_collecthsmetrics
Cancelling job 5089971 - expansionhunter
Cancelling job 5089973 - tiddit
Cancelling job 5089974 - sv_combinevariantcallsets
Cancelling job 5089975 - sv_annotate
Cancelling job 5089979 - sv_reformat
Cancelling job 5089980 - vcf2cytosure_ar
Cancelling job 5089981 - me_merge_bam
Cancelling job 5089982 - me_merge_bam
Cancelling job 5089983 - me_merge_bam
Cancelling job 5089984 - retroseq
Cancelling job 5089985 - retroseq
Cancelling job 5089986 - retroseq
Cancelling job 5089987 - me_merge_vcfs
Cancelling job 5089988 - me_annotate
Cancelling job 5089989 - me_varianteffectpredictor
Cancelling job 5089990 - me_filter
Cancelling job 5089995 - gatk_combinevariantcallsets
Cancelling job 5089996 - peddy_ar
Cancelling job 5089997 - plink
Cancelling job 5089998 - gatk_variantevalall
Cancelling job 5089999 - gatk_variantevalall
Cancelling job 5090000 - gatk_variantevalall
Cancelling job 5090001 - gatk_collectreadcounts
Cancelling job 5090002 - gatk_collectreadcounts
Cancelling job 5090003 - gatk_collectreadcounts
Cancelling job 5090004 - gatk_denoisereadcounts
Cancelling job 5090005 - gatk_denoisereadcounts
Cancelling job 5090006 - gatk_denoisereadcounts
Cancelling job 5090007 - gens_generatedata
Cancelling job 5090008 - gens_generatedata
Cancelling job 5090009 - gens_generatedata
Cancelling job 5090014 - rhocall_viz
Cancelling job 5090015 - rhocall_viz
Cancelling job 5090016 - rhocall_viz
Cancelling job 5090017 - chromograph_rhoviz
Cancelling job 5090018 - chromograph_rhoviz
Cancelling job 5090019 - chromograph_rhoviz
Cancelling job 5090020 - upd_ar
Cancelling job 5090021 - chromograph_upd
Cancelling job 5090022 - mt_annotation
Cancelling job 5090028 - endvariantannotationblock
Cancelling job 5090029 - gatk_variantevalexome
Cancelling job 5090030 - gatk_variantevalexome
Cancelling job 5090031 - gatk_variantevalexome
Cancelling job 5090032 - qccollect_ar
Cancelling job 5090033 - version_collect_ar
Cancelling job 5090034 - multiqc_ar
Cancelling job 5090035 - analysisrunstatus
Cancelling job 5090036 - sacct
Cancelling job 5089934 - bwa_mem2
Cancelling job 5089931 - fastqc_ar
Cancelling job 5089932 - fastqc_ar
Cancelling job 5089933 - fastqc_ar
Case justhusky - Analysis 278296: all ongoing jobs cancelled successfully!
Updated status justhusky - 278296: running
```
Screenshot 2023-09-05 at 08 29 48

**Pull request merged**